src/Sparrow/Json/JsonContextPoolBase.cs

```diff
@@ -91,21 +120,58 @@ private void EnsureCurrentThreadContextWillBeReleased(int currentThreadId)
                     {
                         if(Interlocked.CompareExchange(ref copy[i].ThreadId, currentThreadId, -1) == -1)
                         {
-                            _releaser.ThreadIdHolder = copy[i];
+                            _releaser.ThreadIdHolders.Add(copy[i]);
                             return;
                         }
                         return;
```

Comment:
This `return` needs to be removed

Reply:
I don't like it either, @ayende ?

Reply:
The problem is that the `for` loop will not work as expected. When we fail to do compareExchange on first item, then we are returning and not registering our thread at all.

Reply:
done